server/routerlicious/packages/lambdas/src/scribe/checkpointManager.ts

```diff
@@ -136,19 +140,85 @@ export class CheckpointManager implements ICheckpointManager {
 		}
 	}
 
-	private async writeScribeCheckpointState(checkpoint: IScribe) {
-		await this.documentRepository.updateOne(
-			{
-				documentId: this.documentId,
-				tenantId: this.tenantId,
-			},
-			{
-				// MongoDB is particular about the format of stored JSON data. For this reason we store stringified
-				// given some data is user generated.
-				scribe: JSON.stringify(checkpoint),
-			},
-			null,
-		);
+	private async writeScribeCheckpointState(checkpoint: IScribe, noActiveClients: boolean) {
+		const lumberProperties = getLumberBaseProperties(this.documentId, this.tenantId);
+
+		if (this.localCheckpointEnabled) {
+			if (
+				this.localCheckpointCollection !== null &&
+				this.localCheckpointCollection !== undefined
+			) {
+				if (!noActiveClients) {
+					// If there are active clients, we write to the local collection
+					Lumberjack.info(`Writing checkpoint to local database`, lumberProperties);
+					// Use upsert in case document does not exist in the local database
+					await this.writeScribeCheckpointToCollection(checkpoint, true);
+				} else {
+					// No active clients, delete from local collection, write to global collection
+					Lumberjack.info(
+						`Removing checkpoint data from the local database. No active clients.`,
+						lumberProperties,
+					);
+					await Promise.all([
+						this.localCheckpointCollection
+							.deleteOne({
+								documentId: this.documentId,
+								tenantId: this.tenantId,
+							})
+							.catch((error) => {
+								Lumberjack.error(
+									`Error removing checkpoint data from the local database.`,
+									lumberProperties,
+									error,
+								);
+							}),
+						await this.writeScribeCheckpointToCollection(checkpoint, false),
+					]);
+				}
+			} else {
+				Lumberjack.info(
+					`No local collection found. Writing checkpoint to global database.`,
+					lumberProperties,
+				);
+				await this.writeScribeCheckpointToCollection(checkpoint, false);
+			}
+		} else {
+			Lumberjack.info(`Local checkpoints disabled. Writing checkpoint to global database.`);
+			await this.writeScribeCheckpointToCollection(checkpoint, false);
+		}
+	}
+
+	private async writeScribeCheckpointToCollection(checkpoint: IScribe, isLocal: boolean = false) {
+		const lumberProperties = getLumberBaseProperties(this.documentId, this.tenantId);
+		const checkpointFilter = {
+			documentId: this.documentId,
+			tenantId: this.tenantId,
```

Comment:
For every Write, you will have to have `_id`. That is why I suggest you have a `checkpointRepository`. So you can write function like `writeCheckpointRepository(documentId: string, tenantId: string, updates: any)`''

Remember, read/write have to both have `_id` or don't have `_id`, otherwise you will not able to read data properly. Having `_id` supports you the point operation power


You don't need to pass `_id` to `documentRepository` because `documentRepository` take care of `_id` in FRS repo